### PR TITLE
[Feature]: Heatmap 2way sync

### LINKED ISSUE #1011: [2-WAY COMMUNICATION]: Heatmap chart
2 way communication for Heatmap chart as per https://github.com/opensearch-project/observability/pull/987

@abasatwar @spattnaik

### PR DESCRIPTION
**Pull request by DipraAich**
### Description
This PR includes 2 way sync for Heatmap.

### Issues Resolved
#1011

### Check List
- [ ] New functionality includes 2way sync for Heatmap.
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by abasatwar — APPROVED** [review]

File: dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx [hunk 1]
```diff
@@ -37,7 +37,9 @@ export const HeatMap = ({ visualizations, layout, config }: any) => {
     isEmpty(zMetrics) ||
     isEmpty(data[xaxisField.label]) ||
     isEmpty(data[yaxisField.label]) ||
-    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`])
+    isEmpty(data[`${zMetrics.aggregation}(${zMetrics.name})`]) ||
+    dataConfig?.dimensions.length > 2 ||
```
Comment (mengweieric) [inline dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx:41]: what if only 1 dimension is selected?

Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx:41]: If less than 2 dimension are selected the Heatmap doesn't render. That case was already handled.

Comment (mengweieric) [reply dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx:41]: ok cool, any hints on UI to show to the user?

**Review by mengweieric — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

File: dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx (hunk 1 above)
Comment (DipraAich) [reply dashboards-observability/public/components/visualizations/charts/maps/heatmap.tsx:41]: image

This is how it currently shows in the UI

**Pull request merged**